### PR TITLE
[AIRFLOW-6089] Reorder setup.py dependencies and add ci

### PR DESCRIPTION
**Pull request by zhongjiajie**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6089
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

reorder setup package in alphabetical and add unittest to ci

### Tests

- [x] My PR adds the following unit tests :

Add test

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Comment by zhongjiajie** [comment]
CI passed in https://travis-ci.org/apache/airflow/builds/617797191?utm_source=github_status&utm_medium=notification

**Comment by zhongjiajie** [comment]
I'm not sure if CI work , so I add a failed commit to test if static test work.

**Comment by zhongjiajie** [comment]
> I'm not sure if CI work , so I add a failed commit to test if static test work.

Seem Travis-CI pre-test stage don't catch the error, but test stage catch https://travis-ci.org/apache/airflow/builds/617989011?utm_source=github_status&utm_medium=notification

Have to more work on it

File: scripts/ci/_utils.sh [hunk 1]
```diff
@@ -648,7 +648,7 @@ function run_check_setup() {
             --env HOST_GROUP_ID="$(id -gr)" \
             --rm \
             "${AIRFLOW_CI_IMAGE}" \
-            "--" "/opt/airflow/scripts/ci/in_container/run_check_licence.sh" \
+            "--" "/opt/airflow/scripts/ci/in_container/run_check_setup.sh" \
```
Comment (zhongjiajie) [inline scripts/ci/_utils.sh]: Found out that wrong script name we call.

**Comment by zhongjiajie** [comment]
Now, Travis-CI failed as we except in https://travis-ci.org/apache/airflow/builds/618058541?utm_source=github_status&utm_medium=notification
**BUT NOT SHOW ERROR DETAIL**

**Comment by potiuk** [comment]
Hey @zhongjiajie -> It's super cool change. I have one suggestion though. Your test does not require full airflow set of dependencies to run - it's rather standalone, so you can run it as a "python" pre-commit check - similar to isort. Then you would not need all the scripts (they are only needed if you want to run stuff inside docker container).

something like that could work:

```
      - id: setup-order
        name: Checks for an order of dependencies in setup.py
        language: python
        files: ^setup.py$
        pass_filenames: false
        require_serial: true
        entry: tests/test_order_setup.py
```

It will automatically create a separate virtualenv for this check and run the test in this virtualenv.

**Comment by zhongjiajie** [comment]
@potiuk Thanks for your advices, actually I try to understand how Airflow Breeze work during those days.

**Comment by zhongjiajie** [comment]
pre-commit configure work with 

```sh
      - id: setup-order
        name: Checks for an order of dependencies in setup.py
        language: system
        files: ^setup.py$
        pass_filenames: false
        require_serial: true
        entry: "python ./tests/test_order_setup.py"
```

Will create a new commit tomorrow

**Comment by potiuk** [comment]
You can also run chmod +x ./tests/test_order_setup.py and add shebang "#!/usr/bin/env python" at the beginning of the file to make it executable and get it back to 'entry: tests/test_order_setup.py'

@zhongjiajie -> good that you want to learn how breeze works :). BTW. make sure to take a look at the latest updates to the docs done with Technical Writer Elena: https://github.com/PolideaInternal/airflow/blob/more-gsod-improvements/CONTRIBUTING.rst

**Comment by zhongjiajie** [comment]
@potiuk This PR is ready for review, PTAL

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -458,6 +458,8 @@ image built locally):
 ----------------------------------- ---------------------------------------------------------------- ------------
 ``rst-backticks``                     Checks if RST files use double backticks for code.
 ----------------------------------- ---------------------------------------------------------------- ------------
+``setup-order``                       Checks for an order of dependencies in setup.py
+----------------------------------- ---------------------------------------------------------------- ------------
```
Comment (zhongjiajie) [inline CONTRIBUTING.rst:462]: Not sure what **Breeze** field mean and not add * finally

File: BREEZE.rst [hunk 3]
```diff
@@ -886,7 +886,7 @@ This is the current syntax for  `./breeze <./breeze>`_:
   -F, --static-check-all-files <STATIC_CHECK>
           Run selected static checks for all applicable files. You should specify static check that
           you would like to run or 'all' to run all checks. One of
-          [ all all-but-pylint check-hooks-apply check-merge-conflict check-executables-have-shebangs check-xml debug-statements detect-private-key doctoc end-of-file-fixer flake8 forbid-tabs insert-license check-apache-license lint-dockerfile mixed-line-ending mypy pylint shellcheck].
+          [ all all-but-pylint check-apache-license check-executables-have-shebangs check-hooks-apply check-merge-conflict check-xml debug-statements doctoc detect-private-key end-of-file-fixer flake8 forbid-tabs insert-license lint-dockerfile mixed-line-ending mypy pylint setup-order shellcheck].
```
Comment (zhongjiajie) [inline BREEZE.rst:889]: I make this in alphabetical, same as others hint

Comment (potiuk) [reply BREEZE.rst:889]: Cool!

File: CONTRIBUTING.rst (hunk 2 above)
Comment (potiuk) [reply CONTRIBUTING.rst:462]: It's explained above (and we have a change in progress that it gets explained better). It's exactly the difference whether it uses Docker Breeze image underneath or not. Your test initially used it (with the scripts etc.) but then now with "python" it does not use those images. So it's ok as it is.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Cool! very nice. Thanks @zhongjiajie !

**Comment by potiuk** [comment]
I also cherry-picked it to v1-10-test :)

**Comment by zhongjiajie** [comment]
Thanks Jarek's review and merge

**Pull request merged**